### PR TITLE
Feature/sprint 15

### PR DESCRIPTION
**Pull request by hardik-desai-kcl**

### CODE REVIEW

File: src/Sfa.Tl.Matching.Application.IntegrationTests/BankHoliday/CalendarHttpClient.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Reflection;
+
+namespace Sfa.Tl.Matching.Application.IntegrationTests.BankHoliday
+{
```
Comment (hardik-desai-kcl) [inline src/Sfa.Tl.Matching.Application.IntegrationTests/BankHoliday/CalendarHttpClient.cs]: Raneme this to Fake OR Test CalendarHttpClient

File: src/Sfa.Tl.Matching.Application/Services/ReferralEmailService.cs [hunk 2]
```diff
@@ -285,5 +296,15 @@ await _backgroundProcessHistoryRepository
             return backgroundProcessHistory;
         }
 
+        private static string GetVenueText(string venueName, string venueTown, string venuePostcode, string providerDisplayName)
```
Comment (hardik-desai-kcl) [inline src/Sfa.Tl.Matching.Application/Services/ReferralEmailService.cs]: This should be done in mapper or dto

File: src/Sfa.Tl.Matching.Data/Repositories/OpportunityRepository.cs [hunk 3]
```diff
@@ -192,8 +203,14 @@ where IsValidBasketState(oi, OpportunityType.Referral)
                                           DistanceFromEmployer = re.DistanceFromEmployer,
                                           PlacementsKnown = oi.PlacementsKnown,
                                           Placements = oi.Placements,
-                                          ProviderName = pv.Name == pv.Postcode ? p.DisplayName : $"{pv.Name} (part of {p.DisplayName})",
+                                          ProviderName = ProviderDisplayExtensions.GetDisplayText(pv.Name, pv.Postcode, p.DisplayName),
```
Comment (hardik-desai-kcl) [inline src/Sfa.Tl.Matching.Data/Repositories/OpportunityRepository.cs]: Same as ProviderVenue, put this in to mapper or Dto

Comment (hardik-desai-kcl) [reply src/Sfa.Tl.Matching.Data/Repositories/OpportunityRepository.cs]: That will also simplify the generated SQL

File: src/Sfa.Tl.Matching.Functions/EmployerFeedback.cs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            stopwatch.Stop();
+
+            logger.LogInformation($"Function {context.FunctionName} sent {emailsSent} emails\n" +
+                                  $"\tTime taken: {stopwatch.ElapsedMilliseconds: #,###}ms");
+
+            return new OkObjectResult($"{emailsSent} emails sent.");
+        }
+
+        private async Task<int> SendEmployerFeedbackEmailsAsync(
+            MatchingConfiguration configuration,
```
Comment (hardik-desai-kcl) [inline src/Sfa.Tl.Matching.Functions/EmployerFeedback.cs]: Move these two Methods inside employerFeedbackService.SendEmployerFeedbackEmailsAsync

Comment (hardik-desai-kcl) [reply src/Sfa.Tl.Matching.Functions/EmployerFeedback.cs]: SendEmployerFeedbackEmailsAsync & GetReferralDateAsync

File: src/Sfa.Tl.Matching.Application.IntegrationTests/BankHoliday/CalendarHttpClient.cs (hunk 1 above)
Comment (mikeymonster) [reply src/Sfa.Tl.Matching.Application.IntegrationTests/BankHoliday/CalendarHttpClient.cs]: Renamed in new branch. Also renamed PostcodesIoHttpClient

File: src/Sfa.Tl.Matching.Application/Services/ReferralEmailService.cs (hunk 2 above)
Comment (adamesfa) [reply src/Sfa.Tl.Matching.Application/Services/ReferralEmailService.cs]: Updated

File: src/Sfa.Tl.Matching.Data/Repositories/OpportunityRepository.cs (hunk 3 above)
Comment (adamesfa) [reply src/Sfa.Tl.Matching.Data/Repositories/OpportunityRepository.cs]: Updated

**Pull request merged**